egui/src/widgets/plot/legend.rs

```diff
@@ -1,81 +1,231 @@
-use std::string::String;
+use std::{
+    collections::{BTreeMap, HashSet},
+    string::String,
+};
 
 use crate::*;
 
-pub(crate) struct LegendEntry {
-    pub text: String,
-    pub color: Color32,
-    pub checked: bool,
-    pub hovered: bool,
+use super::items::PlotItem;
+
+/// Where to place the plot legend.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum LegendPosition {
```

Comment:
```suggestion
pub enum Corner {
```

maybe?

Alternatively could use `emath::Align2` here: https://docs.rs/emath/0.12.0/emath/align/struct.Align2.html

Reply:
Good idea to call it `Corner`. I couldn't use `Align2` unfortunately since top, left, right and bottom center are not supported.